core/src/serve_repair.rs

```diff
@@ -637,12 +672,19 @@ impl ServeRepair {
         };
 
         let mut dropped_requests = 0;
-        while let Ok(more) = requests_receiver.try_recv() {
+        let mut well_formed_requests = discard_malformed_repair_requests(&mut reqs_v[0], stats);
+        for mut more in requests_receiver.try_iter() {
             total_requests += more.len();
-            if total_requests > max_buffered_packets {
+            if well_formed_requests > max_buffered_packets {
                 dropped_requests += more.len();
```

Comment:
One thing that's a little funky is if we discard all of the packets after filtering, it will now count them as dropped (whereas if we filter out a bunch of packets but still have at least 1 valid, they don't get counted as dropped).

Also, could get rid of 1 level of nesting with something like:
```
total_requests += more.len();
if well_formed_requests > max_buffered_packets {
    // Already exceeded max. Don't waste time discarding
    dropped_requests += more.len();
    continue;
} 

let retained = discard_malformed_repair_requests(&mut more, stats);
well_formed_requests += retained;
if well_formed_requests > max_buffered_packets {
    dropped_requests += more.len();
    continue;
}  else if retained > 0 {
    reqs_v.push(more);
}
```

Reply:
updated to remove one level of nesting.

The dropped request count more closely resembles the original with the full batch size being counted. Regardless, this count is used for `stats.dropped_requests_load_shed += dropped_requests`. This is useful to see the extent of load shedding. I don't think an exact count is necessary since the count would also be affected by the upstream batching behavior.